libs/fluid-elements/provider/src/lib/provider.ts

```diff
@@ -0,0 +1,232 @@
+
+  /** Styles of the provider component */
+  private _style: CSSStyleDeclaration;
+
+  constructor() {
+    super();
+
+    // Sourced from https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/design-system-provider/design-system-provider.ts#L322
+    if (supportsAdoptedStylesheets && this.shadowRoot !== null) {
+      const sheet = new CSSStyleSheet();
+      sheet.insertRule(':host{}');
+      (this.shadowRoot as any).adoptedStyleSheets = [
+        ...(this.shadowRoot as any).adoptedStyleSheets,
+        sheet,
+      ];
+      this._style = (sheet.rules[0] as CSSStyleRule).style;
+    } else {
+      // Fallback if the browser doesn't support adopted stylesheets
+      // See https://caniuse.com/#feat=mdn-api_documentorshadowroot_adoptedstylesheets
+      this._style = this.style;
+    }
+  }
+
+  /** Update lifecycle */
+  update(props: Map<string | number | symbol, unknown>): void {
+    this._style.cssText = this._renderDynamicStyle().toString();
+    super.update(props);
+  }
+
+  /**
+   * Render function of the custom element. It is called when one of the
+   * observedProperties (annotated with @property) changes.
+   */
+  render(): TemplateResult {
+    return html` <slot></slot> `;
+  }
+
+  /** These styles are generated dynamically based on the component attributes */
+  private _renderDynamicStyle(): CSSResult {
+    // Helpers to keep this code cleaner
+    const token = this._getTokenForCSS.bind(this);
+    const spacing = this._getSpacingForCSS.bind(this);
+    const color = this._getColorForCSS.bind(this);
+
+    return css`
+      /* Legibility definitions */
+      -webkit-font-smoothing: antialiased;
+      -moz-osx-font-smoothing: grayscale;
+      text-rendering: optimizeLegibility;
+
+      /* Globals */
+      --fluid-layout--multiplicator: ${token(
+        `FLUID_LAYOUT_${this.layout.toUpperCase()}` as any,
+      )};
+
+      /* Spacing */
+      --fluid-spacing--0: ${spacing('FLUID_SPACING_0')};
+      --fluid-spacing--3x-small: ${spacing('FLUID_SPACING_3X_SMALL')};
+      --fluid-spacing--2x-small: ${spacing('FLUID_SPACING_2X_SMALL')};
+      --fluid-spacing--x-small: ${spacing('FLUID_SPACING_X_SMALL')};
+      --fluid-spacing--small: ${spacing('FLUID_SPACING_SMALL')};
+      --fluid-spacing--medium: ${spacing('FLUID_SPACING_MEDIUM')};
+      --fluid-spacing--large: ${spacing('FLUID_SPACING_LARGE')};
+      --fluid-spacing--x-large: ${spacing('FLUID_SPACING_X_LARGE')};
+      --fluid-spacing--2x-large: ${spacing('FLUID_SPACING_2X_LARGE')};
+      --fluid-spacing--3x-large: ${spacing('FLUID_SPACING_3X_LARGE')};
+
+      /* Colors */
+      --color-background: ${color('COLOR_BACKGROUND')};
+      --color-maxcontrast: ${color('COLOR_MAXCONTRAST')};
+      --color-primary-70: ${color('COLOR_PRIMARY_70')};
+      --color-primary-80: ${color('COLOR_PRIMARY_80')};
+      --color-primary-90: ${color('COLOR_PRIMARY_90')};
+      --color-primary-100: ${color('COLOR_PRIMARY_100')};
```

Comment:
this seems like we will forget to add variables here at some point :D Any ideas on how to make this a bit more failsafe while maintaining.

Reply:
https://github.com/dynatrace-oss/barista/blob/85586adad5a49564232eb621ef8ba2ff5aa8d60a/libs/fluid-elements/provider/src/lib/provider.spec.ts#L88 :)